AppCenter/AppCenter/Internals/Storage/MSStorageBindableType.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Defines the storage type to be bound in an sql statement.
+ */
+@protocol MSStorageBindableType <NSObject>
+
+@required
+
+/**
+ * Binds itself with a statement.
+ *
+ * @param query sqlite statement.
+ * @param index position of the parameter.
+ *
+ * @return int result code.
+ */
+- (int)bindWithStatement:(void *)query atIndex:(int)index;
```

Comment:
Please consider using `NSUInteger` instead of `int`.

Reply:
Why would we do that? `sqlite_bind` would not accept it and we'll have a cast anyway.

Reply:
I see.